app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -1958,6 +1960,113 @@ public function setUtmTags(Lead $lead, UtmTag $utmTags)
     }
 
     /**
+     * Add leads UTM tags via API.
+     *
+     * @param Lead  $lead
+     * @param array $query
+     */
+    public function addUTMTags(Lead $lead, $query)
+    {
+        // available fields and it's setter.
+        $fields = ['utm_campaign' => 'UtmCampaign',
+                   'utm_source'   => 'UtmSource',
+                   'utm_medium'   => 'UtmMedium',
+                   'utm_content'  => 'UtmContent',
+                   'utm_term'     => 'UtmTerm',
+                   'useragent'    => 'UserAgent',
+                   'url'          => 'Url',
+                   'referer'      => 'Referer',
+                   'query'        => 'Query',
+                   'remotehost'   => 'RemoteHost',
+                   'lastActive'   => 'DateAdded',
+        ];
+
+        // 'query' must be an array
```

Comment:
Actually the `$query` parameter is not well named! the 'query' in that comment was actually the `$query['query']` parameter. So just in case it's provided as a html query string (`a=b&b=1`) - which needs to be url escaped - I check and convert it to a query Array.